crates/file_store/src/store.rs

```diff
@@ -25,16 +24,17 @@ impl<C> PersistBackend<C> for Store<C>
 where
     C: Append + serde::Serialize + serde::de::DeserializeOwned,
 {
-    type WriteError = std::io::Error;
-
-    type LoadError = IterError;
-
-    fn write_changes(&mut self, changeset: &C) -> Result<(), Self::WriteError> {
-        self.append_changeset(changeset)
+    fn write_changes(&mut self, changeset: &C) -> anyhow::Result<()> {
+        self.append_changeset(changeset).map_err(|error| {
+            anyhow::Error::msg(error.to_string())
+                .context("Failed to write changes to the persistence backend.")
+        })
     }
 
-    fn load_from_persistence(&mut self) -> Result<Option<C>, Self::LoadError> {
-        self.aggregate_changesets().map_err(|e| e.iter_error)
+    fn load_from_persistence(&mut self) -> anyhow::Result<Option<C>> {
+        self.aggregate_changesets().map_err(|e| {
+            anyhow::Error::msg(e.to_string()).context("Error loading from persistence backend.")
+        })
```

Comment:
We still want to map to `iter_error`.

```suggestion
        self.aggregate_changesets()
            .map_err(|e| anyhow!(e.iter_error).context("error loading from persistence backend"))
```